### PR TITLE
Updated version to v0.9.0

### PR DESCRIPTION
**Pull request by scotthain**
dictdiffer v0.9.0

**Destination channel:** defaults

### Links

- [PKG-5695](https://anaconda.atlassian.net/browse/PKG-5695) 
- [dictdiffer](https://github.com/inveniosoftware/dictdiffer)

### Explanation of changes:
- Updated version & sha

[PKG-5695]: https://anaconda.atlassian.net/browse/PKG-5695?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,7 +13,8 @@ source:
 build:
   noarch: python
   number: 0
-  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vvv "
+  skip: True # [py<35]
```
Comment (scotthain) [inline recipe/meta.yaml]: Should this be <37?

Comment (scotthain) [reply recipe/meta.yaml]: Yes it should :)

**Review by bwingconda — DISMISSED** [review]
LGTM. Could run their unit tests as well.

**Review by Arishamays1 — COMMENTED** [review]
Let's add pip back to the host section

File: recipe/meta.yaml [hunk 2]
```diff
@@ -11,22 +11,24 @@ source:
...
+  skip: True # [py<37]
+  script: "{{ PYTHON }} -m pip install . --no-deps --no-build-isolation --ignore-installed -vvv"
 
 requirements:
   host:
-    - pip
     - python
     - pytest-runner >=2.7
     - setuptools_scm >=3.1
+    - wheel
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:25]:
```suggestion
    - setuptools
    - wheel
```
Missing `setuptools` as a Python build backend, see https://github.com/inveniosoftware/dictdiffer/blob/v0.9.0/setup.py#L17. We explicitly add it to `host` in such cases.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -11,22 +11,24 @@ source:
   sha256: "{{ sha256 }}"
 
 build:
-  noarch: python
   number: 0
-  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vvv "
+  skip: True # [py<37]
+  script: "{{ PYTHON }} -m pip install . --no-deps --no-build-isolation --ignore-installed -vvv"
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Double quotes can be removed as the YAML key is already a string. The same applies to other places in the recipe.
The verbosity `-vvv` can be reduced to `-vv` for almost all Python packages

**Review by skupr-anaconda — DISMISSED** [review]
LGTM but the YAML style and `setuptools`

File: recipe/meta.yaml [hunk 4]
```diff
@@ -11,22 +11,24 @@ source:
...
 
 requirements:
   host:
-    - pip
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:20]: `pip` is required to run `pip install .` for building the package

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Pull request merged**